src/main/java/com/metamx/common/parsers/JSONParser.java

```diff
@@ -68,23 +68,33 @@ public Object apply(JsonNode node)
   private final ObjectMapper objectMapper;
   private ArrayList<String> fieldNames;
   private final Set<String> exclude;
+  private final JSONParserConfig parserConfig;
 
   public JSONParser()
   {
     this(new ObjectMapper(), null, null);
   }
 
+  public JSONParser(ObjectMapper objectMapper, JSONParserConfig parserConfig)
+  {
+    this.objectMapper = objectMapper;
+    this.exclude = Sets.newHashSet();
+    this.parserConfig = parserConfig;
+  }
+
   @Deprecated
   public JSONParser(Iterable<String> fieldNames)
   {
     this(new ObjectMapper(), fieldNames, null);
   }
 
+  @Deprecated
```

Comment:
I actually think we should rework the constructor to be
Objectmapper, fieldNames, jsonparseconfig

I don't think excludes should be at this level, or if they are, be in the jsonparseconfig